test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,576 @@
+// This script is automation of Test H24 procedure - to verify self-managed APIcast API gateway deployment
+// Test does not include creation of custom policy
+// Procedure: ./test-cases/tests/products/h24-verify-selfmanaged-apicast-and-custom-policy.md
+// Doc: https://access.redhat.com/documentation/en-us/red_hat_openshift_api_management/1/topic/a702e803-bbc8-47af-91a4-e73befd3da00
+// This test case should prove that it is possible for customers to deploy self-managed APIcast
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	appsv1alpha1 "github.com/3scale/apicast-operator/pkg/apis/apps/v1alpha1"
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	routev1 "github.com/openshift/api/route/v1"
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"golang.org/x/term"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	log                                       = l.NewLogger()
+	threeScaleNamespace                       = "redhat-rhoam-3scale"
```

Comment:
RHMI was not in requirements, but if needed, we will implement in new version of the script (will open task).

Reply:
That's okay, I think we wont use it in the case of RHMI. I think it's fine here but in general it's good to use the prefix from the cr.spec.NamespacePrefix in case this test is used against a different environment which uses a different prefix. For example rhoam-internal or multitentant-managed-api type installation.

Reply:
TODO in https://issues.redhat.com/browse/MGDAPI-3037

Reply:
Done/changed to be similar to test22 - namespace prefix placed to test.sh.  Maybe need assistance to get it from CR; will try to clear  in  https://issues.redhat.com/browse/MGDAPI-3037